feature-set/src/lib.rs

```diff
@@ -0,0 +1,1362 @@
+#![cfg_attr(feature = "frozen-abi", feature(min_specialization))]
+
+use {
+    ahash::{AHashMap, AHashSet},
+    solana_epoch_schedule::EpochSchedule,
+    solana_feature_set_interface::FeatureSet as SdkFeatureSet,
+    solana_hash::Hash,
+    solana_pubkey::Pubkey,
+    solana_sha256_hasher::Hasher,
+    std::sync::LazyLock,
+};
+
+#[cfg_attr(feature = "frozen-abi", derive(solana_frozen_abi_macro::AbiExample))]
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct FeatureSet {
+    feature_set: SdkFeatureSet,
+}
+
+impl Default for FeatureSet {
+    fn default() -> Self {
+        // All features disabled
+        Self {
+            feature_set: SdkFeatureSet {
+                active: AHashMap::new(),
+                inactive: AHashSet::from_iter((*FEATURE_NAMES).keys().cloned()),
+            },
+        }
+    }
+}
+
+impl FeatureSet {
+    pub fn new(active: AHashMap<Pubkey, u64>, inactive: AHashSet<Pubkey>) -> Self {
+        Self {
+            feature_set: SdkFeatureSet { active, inactive },
+        }
+    }
+
+    pub fn active(&self) -> &AHashMap<Pubkey, u64> {
+        &self.feature_set.active
+    }
```

Comment:
Just so I'm sure I understand how this will be used later, is the idea to pass the hashmap directly to sdk users? Ie `verify` will take in a hashmap of active features and a hashmap of inactive features? https://github.com/anza-xyz/solana-sdk/blob/f6af0f597a02785e5ceb9f02cf670d067161afc5/precompiles/src/lib.rs#L43

We might also need a function that exposes the underlying `SdkFeatureSet`, so that the sdk can accept the whole feature set at once

Reply:
nah they're going to consume this directly for awhile as we eliminate all usage of FeatureSet in sdk. then this will go private api again

Reply:
Sorry, I still don't understand. What will the sdk consume directly, and how? If the sdk will use the inner sdk feature set, we'll need to expose the sdk feature set through an API in agave-feature-set.

Reply:
`agave_feature_set::FeatureSet` will be _temporarily_ public api